pkg/gce-pd-csi-driver/controller.go

```diff
@@ -609,7 +609,11 @@ func (gceCS *GCEControllerServer) CreateSnapshot(ctx context.Context, req *csi.C
 			return nil, status.Error(codes.Internal, fmt.Sprintf("Unknown get snapshot error: %v", err))
 		}
 		// If we could not find the snapshot, we create a new one
-		snapshot, err = gceCS.CloudProvider.CreateSnapshot(ctx, project, volKey, req.Name)
+		parameters := make(map[string]string)
+		if req.GetParameters() != nil {
+			parameters = req.GetParameters()
+		}
+		snapshot, err = gceCS.CloudProvider.CreateSnapshot(ctx, project, volKey, req.Name, parameters[common.ParameterStorageLocations])
```

Comment:
We should split, trim and validate the storage locations here (via a helper ftn of course). See for example common.ExtractAndDefaultParameters and its usages as a model.

Reply:
I was thinking about the same thing initially but 

1. It's debatable whether we should validate the regions themselves - the PD backend may change.
2.  If (1) is true, then there is no other validation, because every string can be treated as a comma-separated string.

Reply:
discussed offline. The API takes a list of strings, so we definitely need to split on commas, so we should do that early. If we're splitting on commas, we need to trim whitespace, which means we should do a regexp validation of the location to make sure it's plausible. Things like validating the regions will be done over in the PD service anyway and not in the driver, even in the cloud provider.